### PR TITLE
feat(artwork): Add new countdown timer and label for cascading end time interval sales

### PR DESCRIPTION
**Pull request by annacarey**
This PR resolves [BX-210](https://artsyproduct.atlassian.net/browse/BID-210)

### Description

This PR adds a new timer to cascading end time lots that is formatted differently. This PR also changes the word "Ends" to "Closes" for OO sales that are currently open.

Sale is open and closes in < 2 mins
Screen Shot 2022-04-13 at 12 46 30 PM

Sale is open and closes in < 60 mins 
Screen Shot 2022-04-07 at 12 21 48 PM

Sale is open and closes in <24 hours and >= 1 hour
Screen Shot 2022-04-13 at 12 35 15 PM

Sale is open and closes in >= 24 hours
Screen Shot 2022-04-13 at 12 32 37 PM

Sale opens in < 1 day:
Screen Shot 2022-04-13 at 12 28 10 PM

Sale opens in >=1 day <2 days
Screen Shot 2022-04-13 at 12 30 46 PM

Sale opens in >= 2 days
Screen Shot 2022-04-13 at 12 30 05 PM

### PR Checklist (tick all before merging)

- [ ] Feature flag around my changes.
- [ ] App state migration.
- [ ] Screenshots/videos of my changes.
- [ ] Tested on **iOS** and **Android**.
- [ ] Tests/stories for my changes.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to run this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

- Display a new countdown on lots that are in sales with cascading end times

#### Android user-facing changes

-

#### Dev changes

-

</details>

### CODE REVIEW

File: src/app/Scenes/Artwork/Components/CommercialInformation.tsx [hunk 1]
```diff
@@ -37,11 +38,14 @@ export const CommercialInformationTimerWrapper: React.FC<CommercialInformationPr
...
       isPreview,
       isClosed,
+      cascadingEndTimeInterval,
       liveStartAt: liveStartsAt,
       startAt: startsAt,
-      endAt: endsAt,
+      endAt: saleEndAt,
     } = props.artwork.sale || {}
 
+    const { endAt: lotEndAt } = props.artwork.saleArtwork
```
Comment (annacarey) [inline src/app/Scenes/Artwork/Components/CommercialInformation.tsx:47]: TS was complaining that `endAt` doesn't exist on the `saleArtwork` type but I'm not sure why because it does exist on `CommercialInformation_artwork`

File: .secrets.baseline [hunk 2]
```diff
@@ -418,14 +418,14 @@
         "filename": "src/app/Scenes/Artwork/Components/CommercialInformation.tests.tsx",
         "hashed_secret": "ead2477c86563400ffd2577338c6c83ae48aeb5a",
         "is_verified": false,
-        "line_number": 291
+        "line_number": 298
```
Comment (annacarey) [inline .secrets.baseline:421]: Is it normal to include this in the commit?

Comment (pvinis) [reply .secrets.baseline:421]: yea. totally normal. I can add more context but it's perfectly fine 👍

File: src/app/Components/Bidding/Components/Timer.tsx [hunk 3]
```diff
@@ -106,11 +107,33 @@ export function currentTimerState({ isPreview, isClosed, liveStartsAt }: Props)
...
+  cascadingEndTimeInterval?: number | null
+}
+
+export const Countdown: React.FC<CountdownProps> = ({
+  duration,
+  label,
+  hasStarted,
+  cascadingEndTimeInterval,
+}) => {
+  console.log("CASACDINGENDTIMINTERVL", cascadingEndTimeInterval)
```
Comment (annacarey) [inline src/app/Components/Bidding/Components/Timer.tsx:121]: `const isCascadingEndTimeIntervalSale = !!casacdingEndTimeInterval`

File: src/app/Components/Countdown/Ticker.tsx [hunk 4]
```diff
@@ -84,3 +84,62 @@ export const SimpleTicker: React.FC<SimpleTickerProps> = ({ duration, separator,
...
+  duration,
+  separator,
+  hasStarted,
+  ...rest
+}) => {
+  const timerCopy = getTimerCopy(duration, hasStarted)
+  return (
+    <Sans {...rest} color={timerCopy.color}>
+      {timerCopy.copy}
+    </Sans>
```
Comment (annacarey) [inline src/app/Components/Countdown/Ticker.tsx:102]: Update to `Text` and update props (no need to pass props for styling)

File: src/app/Components/Countdown/Ticker.tests.tsx [hunk 5]
```diff
@@ -69,3 +69,113 @@ describe("LabeledTicker", () => {
...
+        <Theme>
+          <ModernTicker duration={secondsDuration} hasStarted separator="  " size="5" />
+        </Theme>
+      )
+
+      const timerText = getByText("0m 30s")
+
+      expect(timerText).toBeTruthy()
+      expect(timerText.props.color).toEqual("red100")
+    })
```
Comment (annacarey) [inline src/app/Components/Countdown/Ticker.tests.tsx:179]: Add edge cases to test what happens if `duration` is broken: maybe negative or zero. Also when `hasStarted` is not passed.

Comment (annacarey) [reply src/app/Components/Countdown/Ticker.tests.tsx:179]: (Maybe in cases where `duration` is not correct data, block the rendering of this ticker)

File: src/app/Components/Bidding/Components/Timer.tsx [hunk 6]
```diff
@@ -138,13 +161,13 @@ export const Timer: React.FC<Props> = (props) => {
         CountdownComponent={Countdown}
         onCurrentTickerState={() => {
           const state = currentTimerState(props)
-          const { label, date } = relevantStateData(state, props)
-          return { label, date, state } as any // STRICTNESS_MIGRATION
+          const { label, date, hasStarted } = relevantStateData(state, props)
+          return { label, date, state, hasStarted } as any // STRICTNESS_MIGRATION
```
Comment (annacarey) [inline src/app/Components/Bidding/Components/Timer.tsx:165]: Update types here

File: src/app/Scenes/Artwork/Components/CommercialInformation.tsx [hunk 7]
```diff
@@ -52,23 +56,23 @@ export const CommercialInformationTimerWrapper: React.FC<CommercialInformationPr
...
           onNextTickerState={({ state }) => {
             const nextState = nextTimerState(state as AuctionTimerState, {
               liveStartsAt: liveStartsAt || undefined,
             })
-            const { label, date } = relevantStateData(nextState, {
+            const { label, date, hasStarted } = relevantStateData(nextState, {
               liveStartsAt: liveStartsAt || undefined,
               startsAt: startsAt || undefined,
-              endsAt: endsAt || undefined,
+              endsAt: cascadingEndTimeInterval ? lotEndAt || undefined : saleEndAt || undefined,
```
Comment (annacarey) [inline src/app/Scenes/Artwork/Components/CommercialInformation.tsx:73]: same as above

File: src/app/store/config/features.ts [hunk 8]
```diff
@@ -111,6 +111,12 @@ export const features = defineFeatures({
     description: "Allow linking of social accounts on sign up",
     showInAdminMenu: true,
   },
+  ARDisableCascadingEndTimerLotPage: {
+    readyForRelease: false,
+    description: "Enable cascading end times on the lot page",
```
Comment (annacarey) [inline src/app/store/config/features.ts:116]: I will update this to `Disable`

**Review by bhoggard — APPROVED** [review]

File: Podfile.lock [hunk 9]
```diff
@@ -941,6 +937,13 @@ SPEC REPOS:
     - "UIView+BooleanAnimations"
     - "XCTest+OHHTTPStubSuiteCleanUp"
     - YogaKit
+  https://github.com/artsy/Specs.git:
+    - Aerodramus
```
Comment (brainbicycle) [inline Podfile.lock:941]: can you try these steps to see if we can get rid of this churn?
https://www.willowtreeapps.com/craft/how-to-fix-unexpected-podfile-lock-diffs-after-pod-install

**Review by brainbicycle — APPROVED** [review]
minor change request related to podlike churn, otherwise looks good to me!

File: src/app/store/config/features.ts (hunk 8 above)
Comment (brainbicycle) [reply src/app/store/config/features.ts]: wait should this be disable? so we are defaulting to the new feature is enabled?

Comment (annacarey) [reply src/app/store/config/features.ts]: @pvinis mentioned that there's been some issues with deprecating flags that are default true. For this feature, we want to release this with the new cascade timer turned on so that means we want the disable flag to be off initiallly. (We'd only turn it on if we needed to roll something back.) I realize the double negative is confusing but it sounded like this was a convention that you're moving toward?

Comment (annacarey) [reply src/app/store/config/features.ts]: Also related, @brainbicycle I cannot seem to get this to work: https://app.circleci.com/pipelines/github/artsy/eigen/26056/workflows/27b33fdb-20a9-4d62-862a-ce11bb45c456/jobs/77407

I ran `./scripts/update-echo` and it's not producing a diff. is that expected? Does echo need to be deployed?

File: Podfile.lock (hunk 9 above)
Comment (annacarey) [reply Podfile.lock]: done! thanks for suggesting.

File: src/app/Scenes/Artwork/Components/CommercialInformation.tsx (hunk 1 above)
Comment (annacarey) [reply src/app/Scenes/Artwork/Components/CommercialInformation.tsx:48]: Added additional check on line 36 to resolve type error

File: src/app/Components/Bidding/Components/Timer.tests.tsx [hunk 10]
```diff
@@ -6,8 +6,20 @@ import { Sans, Theme } from "palette"
 import React from "react"
 import "react-native"
 
+import { ModernTicker, SimpleTicker } from "app/Components/Countdown/Ticker"
```
Comment (mzikherman) [inline src/app/Components/Bidding/Components/Timer.tests.tsx:9]: I like the Modern vs. Simpler naming.

File: src/app/Components/Bidding/Components/Timer.tests.tsx [hunk 11]
```diff
@@ -232,3 +244,66 @@ describe("timer transitions", () => {
     expect(getMountedTimerText(timer)).toContain("00d  00h  00m")
   })
 })
+
+describe("Countdown", () => {
+  const duration = moment.duration(36180000)
```
Comment (mzikherman) [inline src/app/Components/Bidding/Components/Timer.tests.tsx:249]: #minor what's this number?

File: src/app/Components/Bidding/Components/Timer.tsx [hunk 12]
```diff
@@ -106,11 +108,28 @@ export function currentTimerState({ isPreview, isClosed, liveStartsAt }: Props)
...
+  hasStarted,
+  cascadingEndTimeInterval,
+}) => {
+  const cascadingEndTimeFeatureEnabled = !useFeatureFlag("ARDisableCascadingEndTimerLotPage")
+
   return (
     <Flex alignItems="center">
-      <SimpleTicker duration={duration} separator="  " size="4t" weight="medium" />
-      <Sans size="2" weight="medium">
+      {cascadingEndTimeFeatureEnabled && cascadingEndTimeInterval ? (
```
Comment (mzikherman) [inline src/app/Components/Bidding/Components/Timer.tsx:126]: looks like the right place to fork the logic like this!

File: src/app/Components/Countdown/Ticker.tests.tsx [hunk 13]
```diff
@@ -69,3 +69,113 @@ describe("LabeledTicker", () => {
     expect(queryAllByText(":")).toHaveLength(3)
   })
 })
+
+describe("ModernTicker", () => {
```
Comment (mzikherman) [inline src/app/Components/Countdown/Ticker.tests.tsx:73]: fantastic tests!

**Review by mzikherman — DISMISSED** [review]
Looks awesome! Very well put together and tested.
